test/modules/core/webgl/classes/query.spec.js

```diff
@@ -1,11 +1,27 @@
 /* eslint-disable max-len, max-statements */
-/* global setInterval, clearInterval */
+/* global requestAnimationFrame */
 import test from 'tape-catch';
-import {pollContext, Query} from 'luma.gl';
+import {Query} from 'luma.gl';
 import util from 'util';
 import GL from '@luma.gl/constants';
 import {fixture} from 'luma.gl/test/setup';
 
+function pollQuery(query, t) {
+  let counter = 0;
+
+  return new Promise((resolve, reject) => {
```

Comment:
Why a promise? I don't see any calls to `resolve` or `reject`, is that intentional?

Idea: If fixed, maybe this function could be added to the official query class to let the user create an adhoc promise for a `Query` (at the expense of running a temporary extra animation loop, but without all the bundle size and code complexity overhead of the query manager)? That would seem like a pretty decent compromise to me!

Reply:
Deal!